www/nginx/src/opnsense/service/templates/OPNsense/Nginx/location.conf

```diff
@@ -164,7 +168,11 @@ location {{ location.matchtype }} {{ location.urlpattern }} {
     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
     proxy_set_header X-Forwarded-Proto $scheme;
     proxy_set_header X-Forwarded-Port $server_port;
+{% if location.proxy_add_port_to_header is defined and  location.proxy_add_port_to_header = '1' %}
+    proxy_set_header X-Forwarded-Host $host:$server_port;
```

Comment:
this header does not support this:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host